### PR TITLE
Issue/332 Rotate-tool fix

### PR DESCRIPTION
**Pull request by Jacco**
### Resolves

Fixes issue #332 concerning text rotation holding shift. Somehow for text items the old code did not work while for other items it did.

### Proposed Changes

The code for shift handling is now simplified and does what it is supposed to do.

Also I changed prevRot to not be an array. It was not needed as an array a number was enough.

### Reason for Changes

Holding shift while rotating seemed to change pivot point while rotating. It was a bug.

### Test Coverage

I did not add tests, but tested rotating several different objects in Chrome.

PS. Sorry for the messy merge PR with the merge commit in it. I know it is possible to make it cleaner and in one commit but I need some advice on how to do it. I probably need to make a new clean fork once all PRs are handled.

### CODE REVIEW

File: src/helper/selection-tools/rotate-tool.js [hunk 1]
```diff
@@ -44,13 +41,9 @@ class RotateTool {
...
                 rotAngle = Math.round(rotAngle / 45) * 45;
-                item.applyMatrix = false;
-                item.pivot = this.rotGroupPivot;
-                item.rotation = rotAngle;
-            } else {
-                item.rotate(rotAngle - this.prevRot[i], this.rotGroupPivot);
             }
-            this.prevRot[i] = rotAngle;
+            item.rotate(rotAngle - this.prevRot, this.rotGroupPivot);
+            this.prevRot = rotAngle;
```
Comment (fsih) [inline src/helper/selection-tools/rotate-tool.js]: I'm seeing an issue where you can only rotate one thing
rotbug

I think you want to bring this line out of the loop

**Review by fsih — CHANGES_REQUESTED** [review]

**Comment by Jacco** [comment]
Good catch. Didn't test that. Code seemed logic :-) Working on it

**Comment by Jacco** [comment]
Hm I dont have that issue it seems :S How do you get the cat in the stand alone scratch-paint?

**Comment by Jacco** [comment]
I do have it now: Its the simplification :P

**Review by fsih — APPROVED** [review]
Looks good, thanks!

**Pull request merged**